const sumRandomNumbers = () => {
  const myArray = Array.from(
    { length: 10 },
    () => Math.floor(Math.random() * 50) + 1
  );
  const sum = myArray.map(number => number * number)
    .reduce((number, acc) => number + acc, 0);

  if (sum >= 8000) {
    throw new Error();
  }
  console.log(sum);
  return sum;
}

const sumArrayFromSum = (sum) => {
  const number = [2, 3, 5, 10].map(number => sum / number).reduce((number, acc) => number + acc);
  console.log(number);
  return number;
}
  
const fetchPromise = async () => {
  try {
    const sum = await sumRandomNumbers();
    const sumFromSum = await sumArrayFromSum(sum);
  } catch (error) {
    console.log('Ã‰ mais de oito mil! Essa promise deve estar quebrada!');
  }
}

fetchPromise();